sbi/inference/potentials/posterior_based_potential.py

```diff
@@ -101,28 +113,43 @@ def __call__(self, theta: Tensor, track_gradients: bool = True) -> Tensor:
                 the potential or manually set self._x_o."
             )
 
-        theta = ensure_theta_batched(torch.as_tensor(theta)).to(self.device)
-
```

Comment:
why is this not needed anymore? especially the `to(self.device)`.

Reply:
you're right, it's still needed. I will add this back